src/wp-includes/option.php

```diff
@@ -956,6 +956,24 @@ function get_transient( $transient ) {
  */
 function set_transient( $transient, $value, $expiration = 0 ) {
 
+	// Transient name should not be empty string, 0, or false.
+	if ( empty( $transient ) ) {
+		return false;
+	}
+
+	// Transient name should be a string or number; bool false is rejected in previous condition.
+	if ( true === $transient || ! is_scalar( $transient ) ) {
+		return false;
+	}
+
+	// Cast transient name as string.
+	$transient = (string) $transient;
+
+	// Check that Transient name is less than or equal to 172 characters.
```

Comment:
An comment explaining the number 172 would be helpful.

Reply:
+1, I left a related comment in https://core.trac.wordpress.org/ticket/58903#comment:6: While I think that it probably makes sense, I feel like enforcing this at the code level is a bit more opinionated than the other checks that this PR adds, so IMO it would be better to discuss separately with other similar cases in core.

Reply:
Yea, this seems to be coming from the DB column: `VARCHAR(191)` which is a "MySQL magic number for VARCHAR columns when innodb_large_prefix is disabled, which is the default for MySQL <= 5.7.6". A comment mentioning that would be pretty helpful imho.

However I'm not sure if 191 depends on the DB charset, i.e. is it for ASKII or it supports UTF-8? In PHP `strlen()` would "cut" UTF-8 characters, afaik. Option names could contain UTF. Thinking this needs a little bit more investigation.

Reply:
Perhaps found a bit more info: "VARCHAR(191) with utf8mb4: 191 is the biggest size that can be fully indexed without enabling innodb_large_prefix". Seems that `strlen()` should be `mb_strlen()`.